scripts/upgrade.sh

```diff
@@ -64,9 +64,29 @@ check_hash(){
     fi
 }
 
-{
+prepare() {
+  MASTER_PLAN=${1}
+  if [ -z "$MASTER_PLAN" ]; then
+    fatal "Master Plan name is not passed to the prepare step. Exiting"
+  fi
+  NAMESPACE=$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)
+  # make sure master plan does exist
+  kubectl get plan $MASTER_PLAN -n $NAMESPACE &>/dev/null || fatal "master plan $MASTER_PLAN doesn't exist"
+  while true; do
+    NUM_NODES=$(kubectl get plan $MASTER_PLAN -n $NAMESPACE -o json | jq '.status.applying | length')
+    if [ "$NUM_NODES" == "0" ]; then
+      break
+    fi
+    info "Waiting for all master nodes to be upgraded"
+    sleep 5
+  done
+}
+
+upgrade() {
   check_hash
   get_k3s_process_info
   replace_binary
   kill_k3s_process
 }
+
+"$@"
```

Comment:
what does this do?

Reply:
This just enable the script to be called with the function name:

./upgrade.sh upgrade
./upgrade.sh prepare